Philips.CodeAnalysis.MaintainabilityAnalyzers/Maintainability/AvoidUnusedToStringAnalyzer.cs

```diff
@@ -0,0 +1,71 @@
+﻿// © 2024 Koninklijke Philips N.V. See License.md in the project root for license information.
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Philips.CodeAnalysis.Common;
+
+namespace Philips.CodeAnalysis.MaintainabilityAnalyzers.Maintainability
+{
+	[DiagnosticAnalyzer(LanguageNames.CSharp)]
+	public class AvoidUnusedToStringAnalyzer : SingleDiagnosticAnalyzer<InvocationExpressionSyntax, AvoidUnusedToStringSyntaxNodeAction>
+	{
+		private const string Title = @"Avoid calling ToString() when the result is discarded";
+		private const string MessageFormat = @"Avoid calling ToString() when the result is discarded or unused";
+		private const string Description = @"Calling ToString() when the result is assigned to a discard variable or used as a standalone expression statement is unnecessary and should be removed.";
+
+		public AvoidUnusedToStringAnalyzer()
+			: base(DiagnosticId.AvoidUnusedToString, Title, MessageFormat, Description, Categories.Maintainability, isEnabled: true)
```

Comment:
New analyzers should be disabled by default (until out-of-band testing has inspired confidence)